Philips.CodeAnalysis.Test/Maintainability/Maintainability/AvoidInvocationAsArgumentTest.cs

```diff
@@ -0,0 +1,318 @@
+﻿// © 2021 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json.Serialization;
+using Philips.CodeAnalysis.Common;
+using Philips.CodeAnalysis.MaintainabilityAnalyzers.Maintainability;
+
+namespace Philips.CodeAnalysis.Test.Maintainability.Maintainability
+{
+	[TestClass]
+	public class AvoidInvocationAsArgumentAnalyzerTest : CodeFixVerifier
+	{
+
+		[TestMethod]
+		public void AvoidInvocationAsArgumentTest()
+		{
+			string template = @"
+class Foo
+{{
+  public Foo() : base(Do()) {}
+  public Foo(string h) {}
+  public static string Do() {{ return ""hi"";}}
+  public string Moo(string s) {{ return ""hi"";}}
+
+  public void MyTest()
+  {{
+    string.Format(Foo.Do());
+    string.Format(nameof(MyTest));
+    string.Format(5.ToString());
+    string.Format(Moo(Do()));	  // Finding
+    Assert.Format(Moo("""").Format(""""));
+  }}
+}}
+";
+			VerifyCSharpDiagnostic(template, DiagnosticResultHelper.Create(DiagnosticIds.AvoidInvocationAsArgument));
```

Comment:
This test is brittle: if the finding moved to another line, this is not detected. Might be better to make separate test cases.

Reply:
Alternatively, assert the Location of the Diagnostic

Reply:
Good call. Added the Location to the Assert.